res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -891,36 +939,74 @@ PioneerDDJSX.crossFaderMSB = function(channel, control, value, status, group) {
 
 PioneerDDJSX.crossFaderLSB = function(channel, control, value, status, group) {
     var fullValue = (PioneerDDJSX.highResMSB[group].crossFader << 7) + value;
-    engine.setValue(group, "crossfader", (fullValue / (0x3FFF / 2)) - 1);
+    engine.setParameter(group, "crossfader", fullValue / 0x3FFF);
 };
 
+
 ///////////////////////////////////////////////////////////////
 //           SINGLE MESSAGE MIDI INPUT HANDLERS              //
 ///////////////////////////////////////////////////////////////
 
 PioneerDDJSX.shiftButton = function(channel, control, value, status, group) {
+    var index = 0,
+        parameter = 0;
     PioneerDDJSX.shiftPressed = (value === 0x7F);
```

Comment:
I don't think it matters much either way. If you want to keep it how it is, to make it work with the Components EffectUnit, you'll need to call the `shift()` and `unshift()` methods for both EffectUnits.